### PR TITLE
Feat/co applicant summary

### PR DESCRIPTION
**Pull request by jonatanhanson**
## Explain the changes you’ve made

Added functionality for co-applicants to display a summary of the case and sign.

## Explain your solution

Added ability for co-applicants to enter the same form as the applicant. The difference is that co-applicants enters the form in "sign mode". This mode locks a case to where the applicant left of, which is on the sign screen. 
Sign mode also disables ability to edit summary fields and hides the form navigation. 

## How to test the changes?

This PR is depending on that encryption works for both applicants and co-applicants. Until then you have to mock the process.
You have to create a case that has applicant and co-applicant and has status: "active:signature:pending". The case must be non-encrypted for the co-applicant.
After that the co-applicant can enter the case summary and sign. 

## Was this feature tested in the following environments?
- [] Storybook on a iOS device/simulator.
- [X] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots

Images of the process:

Screenshot 2021-08-25 at 13 30 58
Screenshot 2021-08-25 at 13 31 19
Screenshot 2021-08-25 at 13 31 45
Screenshot 2021-08-25 at 13 31 52
Screenshot 2021-08-25 at 13 32 05
Screenshot 2021-08-25 at 13 32 15

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-n5dtnq Som medsökande vill jag se sammanfattning för ansökan så att jag kan säkerställa att uppgifterna stämmer före jag signerar](https://app.clickup.com/t/n5dtnq)

File: source/screens/FormCaseScreen.js [hunk 1]
```diff
@@ -18,12 +18,16 @@ const FormCaseScreen = ({ route, navigation, ...props }) => {
   const [form, setForm] = useState(undefined);
   const [formQuestions, setFormQuestions] = useState(undefined);
   const [initialCase, setInitialCase] = useState(undefined);
-  const { caseData, caseId } = route && route.params ? route.params : {};
+  const { caseData, caseId, signMode } = route && route.params ? route.params : {};
```
Comment (KenthLjungHBG) [inline source/screens/FormCaseScreen.js:21]: `signMode` was confusing for me at first. Could we change it to `isSignMode` to keep a boolean naming convention?

File: source/helpers/CaseDataConverter.js [hunk 2]
```diff
@@ -45,9 +45,11 @@ export const convertAnswersToArray = (data, formQuestions) => {
 
   Object.entries(data).forEach((answer) => {
     const [fieldId, value] = answer;
-    const { id, type, tags, ...other } = formQuestions.find((element) => element.id === fieldId);
+    const { id, type, tags, disableValueStorage, ...other } = formQuestions.find(
```
Comment (KenthLjungHBG) [inline source/helpers/CaseDataConverter.js:48]: Where does `disableValueStorage` come from? I can't find any references to it.

**Review by KenthLjungHBG — CHANGES_REQUESTED** [review]
I managed to mock it and it seems to work as intended when triggering from `CaseOverview`, however the same logic is missing from `CaseSummary`. I've also left a few comments 🙂.

File: source/helpers/CaseDataConverter.js (hunk 2 above)
Comment (zilaei) [reply source/helpers/CaseDataConverter.js:48]: disableValueStorage should be created by form builder (as a question) and be used to disable storing values for a corresponding question field/component.
Form builder disableValueStorage key values: https://github.com/helsingborg-stad/mitt-helsingborg-form-builder/commit/3d104a34af6e396f56ec0db0f11a16c2979c1d6e

Comment (KenthLjungHBG) [reply source/helpers/CaseDataConverter.js:48]: Ah I see; my form-builder was out of date 🙂.

**Comment by zilaei** [comment]
Tested the PR, everything works as intended.

**Review by KenthLjungHBG — APPROVED** [review]

**Pull request merged**